hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSEditLogOp.java

```diff
@@ -3826,11 +3826,7 @@ void fromXml(Stanza st) throws InvalidXmlException {
     public String toString() {
       StringBuilder builder = new StringBuilder();
       builder.append("AddCacheDirectiveInfo [");
-      builder.append("id=" + directive.getId() + ",");
-      builder.append("path=" + directive.getPath().toUri().getPath() + ",");
-      builder.append("replication=" + directive.getReplication() + ",");
-      builder.append("pool=" + directive.getPool() + ",");
-      builder.append("expiration=" + directive.getExpiration().getMillis());
+      builder.append(directive);
```

Comment:
What if directive is null?

Reply:
String "null" will be appended to the StringBuilder